bokehjs/src/lib/models/axes/axis.ts

```diff
@@ -46,46 +47,30 @@ export class AxisView extends GuideRendererView {
   panel: Panel
   layout: Layoutable
 
-  private axis_label_math_text_view: MathTextView
-  private major_label_math_text_views: {[key: string]: MathTextView} = {}
+  /*private*/ _axis_label_view: BaseTextView | null = null
+  /*private*/ _major_label_views: Map<string, BaseTextView> = new Map()
 
-  /**
-   * Lazy initialize is called when views are instantiated,
-   * working like an async constructor
-   */
   override async lazy_initialize(): Promise<void> {
     await super.lazy_initialize()
 
-    const {axis_label} = this.model
+    await this._init_axis_label()
+    await this._init_major_labels()
+  }
 
+  protected async _init_axis_label(): Promise<void> {
+    const {axis_label} = this.model
     if (axis_label != null) {
-      if (axis_label instanceof MathText) {
-        this.axis_label_math_text_view = await build_view(axis_label, {parent: this})
-      } else if (is_tex_string(axis_label)) {
-        const math_text = tex_from_text_like(axis_label)
-        if (math_text instanceof TeX) {
-          this.model.axis_label = math_text
```

Comment:
Updating model's properties was one of my concerns. The original would require multiple server updates per plot and would interfere with user's set values.

Reply:
I wouldn't mind splitting `parse_delimited_string` delimited string up too, into a "these are the math parts of the string" part and "turn these math parts into Bokeh models" part, but, later.

Reply:
This will happen next, i.e. composite typeset text and other (math, maybe graphics) content.